modules/apigee_edge_apidocs/src/ApiDocSpecFetcher.php

```diff
@@ -0,0 +1,264 @@
+ */
+class ApiDocSpecFetcher implements ApiDocSpecFetcherInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Drupal\Core\File\FileSystemInterface definition.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * GuzzleHttp\ClientInterface definition.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Drupal\Core\Messenger\MessengerInterface definition.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * The logger.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  private $logger;
+
+  /**
+   * Constructs a new ApiDocSpecFetcher object.
+   *
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   The file_system service.
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The http_client service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger service.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger service.
+   */
+  public function __construct(FileSystemInterface $file_system, ClientInterface $http_client, EntityTypeManagerInterface $entityTypeManager, MessengerInterface $messenger, LoggerInterface $logger) {
+    $this->fileSystem = $file_system;
+    $this->httpClient = $http_client;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->messenger = $messenger;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fetchSpec(ApiDocInterface $apidoc, bool $show_messages = TRUE): bool {
+    $needs_save = FALSE;
+    $spec_value = $apidoc->get('spec')->isEmpty() ? [] : $apidoc->get('spec')->getValue()[0];
+
+    // If "spec_file_source" uses URL, grab file from "file_link" and save it
+    // into the "spec" file field. The file_link field should already have
+    // validated that a valid file exists at that URL.
+    if ($apidoc->get('spec_file_source')->value == ApiDocInterface::SPEC_AS_URL) {
+
+      // If the file_link field is empty, return without changes.
+      if ($apidoc->get('file_link')->isEmpty()) {
+        return FALSE;
+      }
+
+      $file_uri = $apidoc->get('file_link')->getValue()[0]['uri'];
+      $file_uri = Url::fromUri($file_uri, ['absolute' => TRUE])->toString();
+      $request = new Request('GET', $file_uri);
+      $options = [
+        'exceptions' => TRUE,
+        'allow_redirects' => [
+          'strict' => TRUE,
+        ],
+      ];
+
+      // Generate conditional GET header.
+      if (!$apidoc->get('fetched_timestamp')->isEmpty()) {
+        $request = $request->withAddedHeader('If-Modified-Since', gmdate(DateTimePlus::RFC7231, $apidoc->get('fetched_timestamp')->value));
```

Comment:
"GMT" is missing from the end of the value.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since#Syntax

Reply:
`const RFC7231 = 'D, d M Y H:i:s \G\M\T';`

Reply:
Cool, I missed that.